src/common/target-helper.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Target } from 'scanner/iruleresults';
+
+export class TargetHelper {
+    public static getTargetElements = (
+        target: (string | string[])[],
+        dom: Document,
+        targetIndex,
+    ) => {
+        if (!target || target.length < 1) {
+            return;
+        }
+        const selectors = target[targetIndex];
+        let elements: NodeListOf<Element> | null;
+        if (typeof selectors === 'string') {
+            elements = dom.querySelectorAll(selectors);
+        } else {
+            const shadowHost = this.getShadowHost(selectors, dom);
+            if (!shadowHost || !shadowHost.shadowRoot) {
+                return;
+            }
+            elements = shadowHost.shadowRoot.querySelectorAll(selectors[selectors.length - 1]);
+        }
+        return Array.from(elements);
+    };
+
+    public static getTargetElement = (
+        target: (string | string[])[],
+        dom: Document,
+        targetIndex,
+    ) => {
+        if (!target || target.length < 1) {
+            return;
+        }
+        const selectors = target[targetIndex];
+        if (typeof selectors === 'string') {
+            return dom.querySelector(selectors);
+        } else {
+            const shadowHost = this.getShadowHost(selectors, dom);
+            if (!shadowHost || !shadowHost.shadowRoot) {
+                return;
+            }
+            return shadowHost.shadowRoot.querySelector(selectors[selectors.length - 1]);
+        }
+    };
+
+    public static getTargetFromSelector = (selector: string): Target | undefined => {
+        if (selector === '') {
+            return [];
+        }
+        if (!selector) {
+            return;
+        }
+        const selectors: string[] = selector.split(';');
```

Comment:
I don't think this is a new problem introduced by your PR, but on reading this, it's occuring to me that this will break if a scan ever results in a target like `'div[some-data-attr="contains ; or , characters"]'` ([codepen example](https://codepen.io/dbjorge/pen/OJvXaVg), where we can't visualize the automated check that fails on the element in question)